presto-native-execution/presto_cpp/main/operators/BroadcastFactory.cpp

```diff
@@ -83,21 +84,32 @@ void BroadcastFileWriter::collect(const RowVectorPtr& input) {
 
 void BroadcastFileWriter::noMoreData() {}
 
-// TODO: Add file stats - size, checksum, number of rows.
 RowVectorPtr BroadcastFileWriter::fileStats() {
   // No rows written.
   if (numRows_ == 0) {
     return nullptr;
   }
 
-  auto data = BaseVector::create<FlatVector<StringView>>(VARCHAR(), 1, pool_);
-  data->set(0, StringView(filename_));
+  auto fileNameVector =
+      BaseVector::create<FlatVector<StringView>>(VARCHAR(), 1, pool_);
+  fileNameVector->set(0, StringView(filename_));
+  auto maxSerializedSizeVector =
+      BaseVector::create<FlatVector<int64_t>>(BIGINT(), 1, pool_);
+  maxSerializedSizeVector->set(0, maxSerializedSize_);
```

Comment:
QQ: Shouldn't we be reporting back deserializedSize instead? HT in memory will be stored in deserialized format. Right?

Reply:
Had a discussion with @mbasmanova. This is reasonable estimate to row container's size.